src/RestApi/StoreApi/Utilities/ProductQuery.php

```diff
@@ -225,22 +225,22 @@ public function prepare_objects_query( $request ) {
 	}
 
 	/**
-	 * Get objects.
+	 * Get results of query.
 	 *
 	 * @param \WP_REST_Request $request Request data.
 	 * @return array
 	 */
-	public function get_objects( $request ) {
+	public function get_results( $request ) {
 		$query_args = $this->prepare_objects_query( $request );
 
 		add_filter( 'posts_clauses', [ $this, 'add_query_clauses' ], 10, 2 );
 
 		$query       = new \WP_Query();
-		$result      = $query->query( $query_args );
+		$results     = $query->query( $query_args );
 		$total_posts = $query->found_posts;
 
 		// Out-of-bounds, run the query again without LIMIT for total count.
-		if ( $total_posts < 1 ) {
+		if ( $total_posts < 1 && $query_args['paged'] > 1 ) {
```

Comment:
Why this additional condition? Wouldn't we want the logic gated by this condition to run anytime the total retrieved posts is under one for the given query arguments? Or are you intentionally only doing this for paged requests (to account for sites that might not have any posts on the first page for the given query)? 

If this stays, I think it'd be good to make sure it's documented in the squash commit notes when merging this pull.

Reply:
This re-query logic is only for paged results. If you're not paging (returning anything) it's not needed.